sceptre/cli/dump.py

```diff
@@ -48,8 +52,70 @@ def dump_config(ctx, path):
 
     output_format = "json" if context.output_format == "json" else "yaml"
 
+    def process_config(config):
+        stack_name = list(config.keys())[0]
+        file_path = Path(f".dump/{stack_name}/config.yaml")
+        write(config[stack_name], output_format, no_colour=True, file_path=file_path)
+
     if len(output) == 1:
-        write(output[0][stack.external_name], output_format)
+        process_config(output[0])
     else:
         for config in output:
-            write(config, output_format)
+            process_config(config)
+
+
+@dump_group.command(name="template")
```

Comment:
I think I disagree. Sometimes (maybe even most of the time) we'd want to dump the template. But sometimes we'll want to dump the stack-config. 

But I do agree that we shouldn't write (at least by default) both to files and console. I think (in line with every other Sceptre command), we should default to console output. However, you could use `--file` to output to file.

Reply:
I disagree strongly. I never, ever want both, and a change like this would seriously break all our tooling too.

Reply:
So, on thinking about the second part, the problem I see is this. If the command does not always dump to the screen _and_ a file, then we arrive back at the problem that the default behaviour of printing to the screen should also by defaulting tell the user about the stack name. If I do that, then @jfalkenstein 's wish that we can simply redirect a multi-doc yaml to a file is not satisfied. I also like the output better this way, and the dumped files can tell me which template goes with which stack. 

Thoughts?

Reply:
I don't think we ought to dump to a file by default. If there's a `--file` option, though... I don't really care if it goes to the console. I think it would be cleaner if what goes to the console was just "Dumped contents were output to {file path}", though.

Reply:
I agree that console output should be the default and I agree that a multi-doc YAML output makes sense. 

An alternative would be to pipe the rendered content to STDIN and meta information like the stack name to STDERR.